src/objective-c/GRPCClient/GRPCCallOptions.m

```diff
@@ -0,0 +1,431 @@
+                                                    initialMetadata:_initialMetadata
+                                                    userAgentPrefix:_userAgentPrefix
+                                                  responseSizeLimit:_responseSizeLimit
+                                                  compressAlgorithm:_compressAlgorithm
+                                                        enableRetry:_enableRetry
+                                                  keepaliveInterval:_keepaliveInterval
+                                                   keepaliveTimeout:_keepaliveTimeout
+                                                  connectMinTimeout:_connectMinTimeout
+                                              connectInitialBackoff:_connectInitialBackoff
+                                                  connectMaxBackoff:_connectMaxBackoff
+                                              additionalChannelArgs:[_additionalChannelArgs copy]
+                                                        pemRootCert:_pemRootCert
+                                                      pemPrivateKey:_pemPrivateKey
+                                                       pemCertChain:_pemCertChain
+                                                      transportType:_transportType
+                                                   hostNameOverride:_hostNameOverride
+                                                         logContext:_logContext
+                                                  channelPoolDomain:_channelPoolDomain
+                                                          channelId:_channelId];
+  return newOptions;
+}
+
+- (nonnull id)mutableCopyWithZone:(NSZone *)zone {
+  GRPCMutableCallOptions *newOptions = [[GRPCMutableCallOptions allocWithZone:zone]
+      initWithServerAuthority:_serverAuthority
+                      timeout:_timeout
+            oauth2AccessToken:_oauth2AccessToken
+            authTokenProvider:_authTokenProvider
+              initialMetadata:_initialMetadata
+              userAgentPrefix:_userAgentPrefix
+            responseSizeLimit:_responseSizeLimit
+            compressAlgorithm:_compressAlgorithm
+                  enableRetry:_enableRetry
+            keepaliveInterval:_keepaliveInterval
+             keepaliveTimeout:_keepaliveTimeout
+            connectMinTimeout:_connectMinTimeout
+        connectInitialBackoff:_connectInitialBackoff
+            connectMaxBackoff:_connectMaxBackoff
+        additionalChannelArgs:[_additionalChannelArgs copy]
+                  pemRootCert:_pemRootCert
+                pemPrivateKey:_pemPrivateKey
+                 pemCertChain:_pemCertChain
+                transportType:_transportType
+             hostNameOverride:_hostNameOverride
+                   logContext:_logContext
+            channelPoolDomain:_channelPoolDomain
+                    channelId:_channelId];
+  return newOptions;
+}
+
+- (void)setServerAuthority:(NSString *)serverAuthority {
```

Comment:
Here and below you need to copy all NSCopying items to match the declared property behavior

Reply:
Ahh that's very correct. What was I thinking about...